### ISSUE TITLE
False Positive: Layout/ExtraSpacing

### ISSUE THREAD

**Issue by jfelchner** [rubocop-hq/rubocop#5265]
## Reproducible Case

```ruby
# baz.rb
class Foo
  def batch
    @areas = params[:param].map do |ca_params|
               ca_params = ActionController::Parameters.new(stuff)
               event     = Event.find(stuff[:event_id])

               whatever
             end

    render json: @areas
  end
end
```

```yaml
# .rubocop.yml
Layout/ExtraSpacing:
  Enabled:                                    true
  ForceEqualSignAlignment:                    true
```

    > rubocop --only='Layout/ExtraSpacing' baz.rb

`Layout/ExtraSpacing` throws errors on lines 3 and 4 which is incorrect.

## History

As I mentioned in [this comment](https://github.com/bbatsov/rubocop/issues/2478#issuecomment-163988070) back in the day, when checking if there is extra spacing, it should only look at a previous line of _the same indentation level_.  There are three rules which need added to this cop:

## The Three Rules ##

1. If the previous line has greater indentation, disregard this line and instead analyze the one prior to it.  Continue doing this until either Rule 2 or Rule 3 applies.
2. If the previous line has the same indentation, stop and verify.  You're done.
3. If the previous line has less indentation, stop.  There is nothing to verify the line against.  You're done.

Let's look at an example with these three rules:

### A Properly-Aligned Example ###

```ruby
    def batch
      @areas   = params[:param].map {
                   var_1      = 123_456
                   variable_2 = 456_123 }
      @another = params[:param].map {
                   char_1 = begin
                              variable_1_1 = 'a'
                              variable_1_2 = 'b'
                            end
                   var_2  = 456_123 }

      render json: @areas
    end
```

* `var_2` finds the last line with an indentation equal to it _skipping lines which are more indented than it is_
* `variable_1_2` finds the line with an indentation equal to it _and stops because the previous line is_
* `variable_1_1` finds the last line with an indentation equal to it _but stops because the previous line is indented less than it_
* `char_1` finds the last line with an indentation equal to it _but stops because the previous line is indented less than it_
* `@another` finds the last line with an indentation equal to it _skipping lines which are more indented than it is_

etc, etc.

In order to get a "correct" `ExtraSpacing` cop, these rules need to be added.

**Comment by neontapir** [comment]
I tried running the sample method against rubocop 0.52 and wasn't able to reproduce the issue.

**Comment by jfelchner** [comment]
@neontapir This has been busted for over a year, but it looks like some changes were made to `ExtraSpacing` in the last few days.  I don't think the changes solve my issue, but I'll come back and give an offending example if it's still broken.  Closing for now.

**Comment by jfelchner** [comment]
@neontapir I'm assuming you didn't have `ForceEqualSignAlignment` set to `true` in your configuration.

**Comment by jfelchner** [comment]
I've updated the original issue to describe the exact reproducible use case.

**Comment by jfelchner** [comment]
This is still an issue.

**Comment by jfelchner** [comment]
Fixed in #7211
